Lib/ufo2ft/featureWriters/cursFeatureWriter.py

```diff
@@ -0,0 +1,129 @@
+from fontTools.misc.fixedTools import otRound
+
+from ufo2ft.featureWriters import BaseFeatureWriter, ast
+from ufo2ft.featureWriters.kernFeatureWriter import unicodeScriptDirection
+from ufo2ft.util import classifyGlyphs
+
+
+class CursFeatureWriter(BaseFeatureWriter):
+    """Generate a curs feature base on glyph anchors.
+
+    The default mode is 'skip': i.e. if the 'curs' feature is already present in
+    the feature file, it is not generated again.
+
+    The optional 'append' mode will add extra lookups to an already existing
+    features, if any.
+
+    By default, anchors names 'entry' and 'exit' will be used to connect the
+    'entry' anchor of a glyph with the 'exit' anchor of the preceding glyph.
+    """
+
+    tableTag = "GPOS"
+    features = frozenset(["curs"])
+
+    entryAnchorNames = {"entry"}
+    exitAnchorNames = {"exit"}
+
+    def _makeCursiveFeature(self):
+        cmap = self.makeUnicodeToGlyphNameMapping()
+        if any(unicodeScriptDirection(uv) == "LTR" for uv in cmap):
+            gsub = self.compileGSUB()
+            dirGlyphs = classifyGlyphs(unicodeScriptDirection, cmap, gsub)
+            shouldSplit = "LTR" in dirGlyphs
+        else:
+            shouldSplit = False
+
+        lookups = []
+        ordereredGlyphSet = self.getOrderedGlyphSet().items()
+        if shouldSplit:
+            # Make LTR lookup
+            LTRlookup = self._makeCursiveLookup(
+                (
+                    glyph
+                    for (glyphName, glyph) in ordereredGlyphSet
+                    if glyphName in dirGlyphs["LTR"]
+                ),
+                direction="LTR",
+            )
+            if LTRlookup:
+                lookups.append(LTRlookup)
+
+            # Make RTL lookup with other glyphs
+            RTLlookup = self._makeCursiveLookup(
+                (
+                    glyph
+                    for (glyphName, glyph) in ordereredGlyphSet
+                    if glyphName not in dirGlyphs["LTR"]
+                ),
+                direction="RTL",
+            )
+            if RTLlookup:
+                lookups.append(RTLlookup)
+        else:
+            lookup = self._makeCursiveLookup(
+                (glyph for (glyphName, glyph) in ordereredGlyphSet)
+            )
+            if lookup:
+                lookups.append(lookup)
+
+        if lookups:
+            feature = ast.FeatureBlock("curs")
+            feature.statements.extend(lookups)
+            return feature
+
+    def _makeCursiveLookup(self, glyphs, direction=None):
+        statements = self._makeCursiveStatements(glyphs)
+
+        if not statements:
+            return
+
+        suffix = ""
+        if direction == "LTR":
+            suffix = "_ltr"
+        elif direction == "RTL":
+            suffix = "_rtl"
+        lookup = ast.LookupBlock(name=f"curs{suffix}")
+
+        if direction != "LTR":
```

Comment:
this catches `direction=None` too, is that intentional? In that case, why not have `direction="RTL"` by default in the method parameter?